packages/dds/tree/src/test/util/testChangeRebaser.spec.ts

```diff
@@ -0,0 +1,85 @@
+    compose: (changes: number[]) => changes.reduce((a, b) => a + b, 0),
+    invert: (change: number) => -change,
+    rebaseAnchors: (anchor: AnchorSet, over: number) => {},
+});
+
+describe("testChangeRebaser", () => {
+    it("test counter with safe integers", () => {
+        const output = verifyChangeRebaser(counterRebaser, new Set([-1, 2, 3, 0, -2, 4]), (a, b) => a === b);
+        assert.equal(output.rebaseLeftDistributivity, "PASSED");
+        assert.equal(output.composeAssociativity, "PASSED");
+        assert.equal(output.rebaseRightDistributivity, "PASSED");
+        assert.equal(output.rebaseOverDoUndoPairIsNoOp, "PASSED");
+        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "PASSED");
+        assert.equal(output.composeWithInverseIsNoOp, "PASSED");
+    });
+
+    it("test counter with unsafe integers", () => {
+        const output = verifyChangeRebaser(counterRebaser, new Set([Number.MAX_SAFE_INTEGER, -10, 2]), (a, b) => a === b);
+        assert.equal(output.rebaseLeftDistributivity, "PASSED");
+        assert.notEqual(output.composeAssociativity, "PASSED");
+        assert.equal(output.rebaseRightDistributivity, "PASSED");
+        assert.equal(output.rebaseOverDoUndoPairIsNoOp, "PASSED");
+        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "PASSED");
+        assert.equal(output.composeWithInverseIsNoOp, "PASSED");
+    });
+
+    it("test counter of floats with varying number of digits", () => {
+        const output = verifyChangeRebaser(
+            counterRebaser,
+            new Set([1.0, 1.22, -1.222]),
+            (a, b) => a === b,
+        );
+        assert.equal(output.rebaseLeftDistributivity, "PASSED");
+        assert.notEqual(output.composeAssociativity, "PASSED");
+        assert.equal(output.rebaseRightDistributivity, "PASSED");
+        assert.equal(output.rebaseOverDoUndoPairIsNoOp, "PASSED");
+        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "PASSED");
+        assert.equal(output.composeWithInverseIsNoOp, "PASSED");
+    });
+
+    // This test case contains all the different "edge case" numbers
+    it("test counter with special number types", () => {
+        const output = verifyChangeRebaser(
+            counterRebaser,
+            new Set([
+                Number.NaN,
+                Number.MAX_VALUE,
+                Number.MIN_VALUE,
+                Number.POSITIVE_INFINITY,
+                Number.NEGATIVE_INFINITY,
+                Number.MIN_SAFE_INTEGER,
+                Number.MAX_SAFE_INTEGER,
+            ]),
+            (a, b) => a === b,
+        );
+        assert.notEqual(output.rebaseLeftDistributivity, "PASSED");
+        assert.notEqual(output.composeAssociativity, "PASSED");
+        assert.notEqual(output.rebaseRightDistributivity, "PASSED");
+        assert.notEqual(output.rebaseOverDoUndoPairIsNoOp, "PASSED");
+        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "PASSED");
+        assert.notEqual(output.composeWithInverseIsNoOp, "PASSED");
```

Comment:
Now that we've fixed the code (and after fixing the equivalence function to handle `NaN`s), we should get the following results:
```typescript
        assert.equal(output.rebaseLeftDistributivity, "Passed");
        assert.notEqual(output.composeAssociativity, "Passed");
        assert.equal(output.rebaseRightDistributivity, "Passed");
        assert.equal(output.rebaseOverDoUndoPairIsNoOp, "Passed");
        assert.equal(output.rebaseOverUndoRedoPairIsNoOp, "Passed");
        assert.notEqual(output.composeWithInverseIsNoOp, "Passed");
```
In other words, all the checks for rebase should succeed. This is because `counterRebaser` is commutative (commutative = the order of operations don't matter) which means its rebase implementation doesn't need to do anything (and therefore it's always right).

I suggest making a dummy `ChangeRebaser` that returns garbage. Maybe we can work on that together when you get back.